bokeh/charts/_data_adapter.py

```diff
@@ -137,6 +138,13 @@ def validate_values(values):
 
             return values
 
+        elif hasattr(values, '__array__'):
```

Comment:
I remember I had some issues when trying this and playing with conversions from/to DataFrame/Blaze/Arrays, etc.. and into seemed to be doing it right (while other approaches failed). A that moment I thought it was because I was trying to pass uncommon inputs to blaze but I tried that again and doesn't seem the case. I've open [this issue](https://github.com/ContinuumIO/blaze/issues/984) on blaze project to focus the discussion there and understand the problem first.